src/__tests__/components/NavBar/Profile.test.tsx

```diff
@@ -0,0 +1,31 @@
+import { screen } from "@testing-library/react";
+import Profile from "@/components/NavBar/Profile";
+import renderWithMockedProvider from "../../utils";
+
+describe("Profile", () => {
+  const props = {
+    isOpen: true,
+    onOpen: jest.fn(),
+    onClose: jest.fn(),
+    handleLogout: jest.fn(),
+  };
+  it("should render text correctly", () => {
+    renderWithMockedProvider(
+      <Profile
+        isOpen={props.isOpen}
+        onClose={props.onClose}
+        onOpen={props.onOpen}
+        handleLogout={props.handleLogout}
+      />
+    );
+    const MyAccountElement = screen.getByText("My Account");
+    const MyTemplateElement = screen.getByText("My Template");
+    const MyOrderElement = screen.getByText("My Order");
+    const SignOutElement = screen.getByText("Sign Out");
+
+    expect(MyAccountElement).toBeInTheDocument();
+    expect(MyTemplateElement).toBeInTheDocument();
+    expect(MyOrderElement).toBeInTheDocument();
+    expect(SignOutElement).toBeInTheDocument();
```

Comment:
The mock functions "onOpen" and "onClose" also can be tested if you like.
Use the testing library API "userEvent" to fire a hover(there is also a unhover) event and then test if the mock onOpen or onClose "toHaveBeenCalledTimes(1)"